File: javaee/ServletStudy/src/main/java/com/ung8023/foundation/java/ServletExtendsGeneric.java
package com.ung8023.foundation.java;

import javax.servlet.GenericServlet;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import java.io.IOException;

/**
 * 通过继承GenericServlet实现Servlet
 */
public class ServletExtendsGeneric extends GenericServlet {
    @Override
    public void service(ServletRequest req, ServletResponse res) throws ServletException, IOException {
        System.out.println("service");
    }
}

File: javaee/ServletStudy/src/main/java/com/ung8023/foundation/kotlin/ServletExtendsGeneric.kt
package com.ung8023.foundation.kotlin

import javax.servlet.GenericServlet
import javax.servlet.ServletRequest
import javax.servlet.ServletResponse

class ServletExtendsGeneric : GenericServlet() {
    override fun service(req: ServletRequest?, res: ServletResponse?) {
        res?.writer?.write("这是一个Kotlin的GenericServlet")
    }
}